crates/bevy_render/src/camera/camera.rs

```diff
@@ -170,19 +170,30 @@ impl Camera {
     ///  For logic that requires the full logical size of the
     /// [`RenderTarget`], prefer [`Camera::logical_target_size`].
     ///
-    /// Returns `None` if either:
+    /// Returns an error if either:
     /// - the function is called just after the `Camera` is created, before `camera_system` is executed,
     /// - the [`RenderTarget`] isn't correctly set:
     ///   - it references the [`PrimaryWindow`](RenderTarget::Window) when there is none,
     ///   - it references a [`Window`](RenderTarget::Window) entity that doesn't exist or doesn't actually have a `Window` component,
     ///   - it references an [`Image`](RenderTarget::Image) that doesn't exist (invalid handle),
     ///   - it references a [`TextureView`](RenderTarget::TextureView) that doesn't exist (invalid handle).
     #[inline]
-    pub fn logical_viewport_size(&self) -> Option<Vec2> {
-        self.viewport
-            .as_ref()
-            .and_then(|v| self.to_logical(v.physical_size))
-            .or_else(|| self.logical_target_size())
+    pub fn logical_viewport_size(&self) -> Result<Vec2, LogicalViewportSizeError> {
+        let viewport = self.viewport.as_ref();
+        let logical_viewport_size = viewport.and_then(|v| self.to_logical(v.physical_size));
```

Comment:
I think this is probably going to be clearer to just start the `if let` here, rather than storing the extra `None`.

Reply:
Fixed. It did indeed get much better. Was legacy of some earlier bad design decisions